### PR TITLE
bpo-39639: remove suite node

### PR DESCRIPTION
**Pull request by isidentical**
https://bugs.python.org/issue39639

### CODE REVIEW

File: Doc/whatsnew/3.9.rst
Comment (brettcannon) [inline Doc/whatsnew/3.9.rst]:
```suggestion
* The ``Suite`` node from :mod:`ast` has been removed due to no longer being needed.
  (Contributed by Batuhan Taskaya in :issue:`39639`.)
```

**Review by brettcannon — APPROVED** [review]

**Comment by brettcannon** [comment]
I haven't touched the AST in a while, so it would be great if someone else can just double-check me. Otherwise, @isidentical , if no other review comes in by Friday just ping me and I will merge this.

**Review by serhiy-storchaka — APPROVED** [review]

**Review by pablogsal — APPROVED** [review]

File: Misc/NEWS.d/next/Core and Builtins/2020-02-15-15-29-34.bpo-39639.3mqJjm.rst
Comment (pablogsal) [inline Misc/NEWS.d/next/Core and Builtins/2020-02-15-15-29-34.bpo-39639.3mqJjm.rst]:
```suggestion
Remove the ``Suite`` node from the :mod:`ast` module as it's no longer used. Patch by Batuhan Taskaya.
```

File: Parser/Python.asdl
Comment (vstinner) [inline Parser/Python.asdl:6]: I don't know the syntax of this change. Is this change related to the removal of Suite? If not, please revert this change. If you want to push the change anyway, open a separated PR for it :-)

File: Doc/whatsnew/3.9.rst
Comment (vstinner) [inline Doc/whatsnew/3.9.rst]:
```suggestion
* The ``ast.Suite`` node class has been removed due to no longer being needed.
```

**Review by vstinner — COMMENTED** [review]
I suggest to change the PR title to " bpo-39639: Remove ast.Suite node class". It can be done when the PR is merged.

File: Doc/whatsnew/3.9.rst
Comment (vstinner) [reply Doc/whatsnew/3.9.rst]: I'm mostly confused by "node". It's not a node, but a class. Maybe just replace "node" with "node class".

Please also update the NEWS entry.

File: Parser/Python.asdl
Comment (isidentical) [reply Parser/Python.asdl:6]: I thouht it would be less noise if I push this optional styling with this PR.

File: Doc/whatsnew/3.9.rst
Comment (isidentical) [reply Doc/whatsnew/3.9.rst]: Agreed, `ast.AST` subclasses described as `node classes` in the documentation. Will fix

File: Parser/Python.asdl
Comment (vstinner) [reply Parser/Python.asdl:6]: I will not block this PR just for this. It was just a remark. I wrote it before seeing that multiple other core devs already approved the PR. I don't understand if "*" applies to type_ignore or type_ignoes.

Comment (isidentical) [reply Parser/Python.asdl:6]: This is an optional thing, but all usages in the file (and the examples in the ASDL specification) puts type mark (`*`/`?`) after the field type. This version already can be parse-able with the ASDL parser of the CPython but it makes things hard for outside parsers.

File: Misc/NEWS.d/next/Core and Builtins/2020-02-15-15-29-34.bpo-39639.3mqJjm.rst
Comment (vstinner) [inline Misc/NEWS.d/next/Core and Builtins/2020-02-15-15-29-34.bpo-39639.3mqJjm.rst]: Please update also the NEWS entry.

Comment (isidentical) [reply Misc/NEWS.d/next/Core and Builtins/2020-02-15-15-29-34.bpo-39639.3mqJjm.rst]: sure

File: Parser/Python.asdl
Comment (vstinner) [reply Parser/Python.asdl:6]: I see. Let's keep your change in this PR.

**Comment by vstinner** [comment]
@isidentical: Thanks for the doc update, it's now well better ;-) Thanks also for clearing the old dust of from the early age of Python ;)

**Pull request merged**
